lgc/include/lgc/state/Defs.h

```diff
@@ -140,5 +140,6 @@ static const unsigned InternalPerShaderTable = 0x10000001;
 static const unsigned DescRelocMagic = 0xA5A5A500;
 static const unsigned DescRelocMagicMask = 0xFFFFFF00;
 static const unsigned DescSetMask = 0x000000FF;
+static const unsigned DescRelocPushConst = 0xA5A5A600;
```

Comment:
I haven't followed up with this, but let's please not make things worse -- and since your moving of the linking code, we're now in a position to clean this up properly!

The right thing to do here is to extend PAL's Pal::Util::Abi::UserDataMapping. Let's define a range (0x8000000+?) for compiler-internal use, and then just reserve ranges starting from there, with proper thought given to how big those ranges should be (e.g., the descriptor set range should have 32 entries because that's the limit of descriptor sets per pipeline layout that the driver exposes).

Reply:
I have not moved the linker code yet. When I do, I intend to put these constants into a compiler/linker-private ABI file, and I agree we should put more thought into how they're defined. However I would like to do it in a future commit when I move the linker.